web-registry/src/lib/wallet/hooks/useSignArbitrary.tsx

```diff
@@ -0,0 +1,51 @@
+import { MutableRefObject, useCallback } from 'react';
+import WalletConnect from '@walletconnect/client';
+
+import { UseStateSetter } from 'types/react/use-state';
+
+import { chainInfo } from '../chainInfo/chainInfo';
+import { SignArbitraryParams } from '../wallet';
+import { WalletConfig } from '../walletsConfig/walletsConfig.types';
+
+type Props = {
+  walletConfigRef: MutableRefObject<WalletConfig | undefined>;
+  walletConnect?: WalletConnect;
+  setError: UseStateSetter<unknown>;
+};
+
+export const useSignArbitrary = ({
+  walletConfigRef,
+  walletConnect,
+  setError,
+}: Props) => {
+  const signArbitrary = useCallback(
+    async ({ wallet, nonce }: SignArbitraryParams) => {
+      try {
+        if (wallet?.address) {
+          const walletClient = await walletConfigRef.current?.getClient({
+            chainInfo,
+            walletConnect,
+          });
+
+          const signature = await walletClient?.signArbitrary(
+            chainInfo.chainId,
+            wallet.address,
+            JSON.stringify({
```

Comment:
just a nit; it might be helpful to set up some functions for generating this data, i.e. from the registry server:

https://github.com/regen-network/registry-server/blob/dev/server/middleware/keplrStrategy.ts#L8-L26

Reply:
done [1ef84bf](https://github.com/regen-network/regen-web/pull/1814/commits/1ef84bf962935199985f180fd3e91ebf5b323957)